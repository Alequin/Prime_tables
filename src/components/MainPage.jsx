import React from "react";

class MainPage extends React.Component{

  constructor(props){
    super(props)
  }

  render(){
    return (
      <div>
        Main Page
      </div>
    )
  }
}

export default MainPage;
