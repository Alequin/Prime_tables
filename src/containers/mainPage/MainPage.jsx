import React from "react"
import Nav from "./../nav"

class MainPage extends React.Component{

  constructor(props){
    super(props)
    this.onClickGenerateButton = this.onClickGenerateButton.bind(this)
  }

  onClickGenerateButton(inputText){
    console.log(inputText);
  }

  render(){
    return (
      <div className="main-page">
        <Nav onClickButton={this.onClickGenerateButton}/>
        <section className="content-container">

        </section>
      </div>
    )
  }
}

export default MainPage
