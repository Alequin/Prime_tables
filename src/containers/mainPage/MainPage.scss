.main-page{
  height: 100%;
  display: flex;
  flex-direction: column;

  .content-container{
    flex: 1;
    background-color: red;
  }
}
