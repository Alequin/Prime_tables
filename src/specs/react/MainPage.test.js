import React from 'react';
import MainPage from './../../components/mainPage/MainPage.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<MainPage/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
