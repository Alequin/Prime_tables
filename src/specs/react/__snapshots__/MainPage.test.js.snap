// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main-page"
>
  <nav
    className="nav-bar"
  >
    <form>
      <label>
        <h2>
          How many prime numbers would you like to generate?
        </h2>
        <div
          className="align-center-container"
        >
          <input
            className="input"
            onChange={[Function]}
            step=".01"
            type="number"
            value=""
          />
        </div>
      </label>
      <div
        className="align-center-container"
      >
        <input
          className="button"
          onClick={[Function]}
          type="submit"
          value="Generate prime number multiplication table"
        />
      </div>
    </form>
  </nav>
  <section
    className="content-container"
  >
    <section
      className="table-container"
    >
      <div
        className="row"
      >
        <div
          className="cell-container"
          onClick={[Function]}
        >
          _
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          3
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          5
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="cell-container"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          6
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          10
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="cell-container"
          onClick={[Function]}
        >
          3
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          6
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          9
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          15
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="cell-container"
          onClick={[Function]}
        >
          5
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          10
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          15
        </div>
        <div
          className="cell-container"
          onClick={[Function]}
        >
          25
        </div>
      </div>
    </section>
  </section>
</div>
`;
